Support/bin/latextidy.pl

```diff
@@ -197,10 +197,22 @@
 }
 
 # At this point each \label appears on a line of its own.
-# This hack is designed to restore \labels next to \begin, \end and suchlike, so that code folding leaves the \label visible.
-$string =~ s/({[^}]*}[ \t]*)(\n[ \t]*)((\[[^\]]*\][ \t]*)*(\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*)*)\n?[ \t]*/$1$3$2/mg;
-$string =~ s/((\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*)+)(\n[ \t]*)(\\end{[^}]*})/$4$1$5/mg;
-$string =~ s/(\\section.*(\\label[ \t]*(\[[^\]]*\][ \t]*)*{[^}]*}[ \t]*))(\S)/$1\n$4/mg;
+# This hack is designed to restore \labels next to \begin, \end and suchlike
+# so that code folding leaves the \label visible.
+my $option = '(\[[^\]]*\][ \t]*)';
+my $arg = '({[^}]*}[ \t]*)';
+$string =~ s/(?<argument>${arg})
+		(?<newline_then_indent>\n[ \t]*)
+		(?<options_then_labels>${option}*(\\label[ \t]*${option}?${arg})+)
```

Comment:
This possibility is included for compatibility with the [cleveref](http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/contrib/cleveref/cleveref.pdf) package, specifically the facility to override cross-reference types. Personally I would not wish to be deprived of this option.
Popular packages (of which I believe cleveref is one) sometimes redefine LaTeX commands for enhanced optionality. This seems more likely to elicit user objections of "you haven't supported my favourite package" than "you've allowed for, but not enforced, the possibility of a package I never use". The latter sounds a bit unreasonable to me.